### ISSUE TITLE
Compile Error: hot-load is not a helper

### ISSUE THREAD

**Issue by jdaviderb** [lifeart/ember-ast-hot-load#56]
Hello @lifeart I'm trying to use this addon in a project with PODS structure and  I have this error when  I run `ember test`

version:  `"ember-ast-hot-load": "^0.0.76"`

```JS
Compile Error: hot-load is not a helper
```

![image](https://user-images.githubusercontent.com/4649902/51401107-fdeee380-1b17-11e9-9632-1a29df1099b4.png)

PD: I tried exclude the addon from test build but it's still failing

**Comment by jdaviderb** [comment]
@lifeart I changed the npm version  direct to github’s repo and now it works. maybe you should update the npm package

```JSON
  "ember-ast-hot-load": "git+https://github.com/lifeart/ember-ast-hot-load"
```

**Comment by lifeart** [comment]
@jdaviderb can you provide `environment` value?

**Comment by lifeart** [comment]
looks like funnel excluded `addon` files, but `ast` transformation not disabled

**Comment by jdaviderb** [comment]
@lifeart 
```javascript
const environment = EmberApp.env();
console.log('ember env:');
console.log(environment);
```
enviroment:
![image](https://user-images.githubusercontent.com/4649902/51406861-54641e00-1b28-11e9-882c-be214cf5aeff.png)

error:
![image](https://user-images.githubusercontent.com/4649902/51406920-82e1f900-1b28-11e9-83db-5950b550862a.png)

**Comment by lifeart** [comment]
@jdaviderb 
can you add `console.log(config.addonContext);`

before: 

https://github.com/lifeart/ember-ast-hot-load/blob/master/lib/ast-transform.js#L385

it should be located in

`node_modules/ember-ast-hot-load/lib/ast-transform.js`

**Comment by lifeart** [comment]
@jdaviderb so, looks like this path not reached in test mode, so, is log available on line `#373` (if we move statement into it) in `ember test` mode?

**Comment by lifeart** [comment]
do you have `hot-load` helpers, placed manually in app templates?

**Comment by jdaviderb** [comment]
@lifeart not, I don't have hot-helper manually in the templates

![image](https://user-images.githubusercontent.com/4649902/51409774-bd4f9400-1b30-11e9-9117-054b071beb37.png)

![image](https://user-images.githubusercontent.com/4649902/51409948-39e27280-1b31-11e9-81e8-2a4636afd2e0.png)

**Comment by lifeart** [comment]
surprising, looks like it can be an temp files cache issue, because ast transform not running in `test` mode -> `hot-load` helper can't appear in templates. But if templates already compiled in dev mode and ember running with `env=test` it can cause this issue.

**Comment by andrewfan** [comment]
@jdaviderb @lifeart I'm having the same issue and `rm -rf tmp/` does not help

**Comment by lifeart** [comment]
@andrewfan can you please create reproduction repo? I will dig it

**Comment by lifeart** [comment]
@jdaviderb could you check againist https://github.com/lifeart/ember-ast-hot-load/pull/59? 
main changes in https://github.com/lifeart/ember-ast-hot-load/pull/59/files#diff-168726dbe96b3ce427e7fedce31bb0bcR50 (50-67)

**Comment by lifeart** [comment]
let's assume this bug fixed in `0.0.78`, due to #59

**Comment by mweidner037** [comment]
I've been facing this issue with ember-ast-hot-load@2.0.3. It's definitely a caching issue and probably not worth debugging; however, clearing the cache is tricky on newer Ember versions, since they use the system tmp folder instead of `<project-dir>/tmp/`.

My workaround is to run ember tests in a script that first runs:
```zsh
# Tested on M1 Mac
rm -r "$(dirname $(mktemp -u))/${USER}/if-you-need-to-delete-this-open-an-issue-async-disk-cache"
```
Ref: https://github.com/ember-cli/ember-cli/issues/1186#issuecomment-1211291483
